lib/flex_commerce_api/api_base.rb

```diff
@@ -113,6 +114,13 @@ def reconfigure options = {}
       def reload_connection_if_required
         _build_connection(true) if connection_object
       end
+
+      # Allows ApiBase to be used as the class and the real class is then calculated from the type
+      def load(params)
+        return super unless self == FlexCommerceApi::ApiBase
+        klass = JsonApiClient::Utils.compute_type(FlexCommerce, params["type"].singularize.classify)
```

Comment:
`classify` performs `singularize` so we can skip it (just `params["type"].classify`)